File: examples/android/CMeSdkJava/app/src/main/java/msdk/java/sample/db/entity/StructuredMessage.java
package msdk.java.sample.db.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.List;

import msdk.java.messages.StructuredMessage.Response;

@Entity
public class StructuredMessage {
    @PrimaryKey(autoGenerate = true)
    public int id;

    @ColumnInfo(name = "pwdid")
    public String pwDid;

    @ColumnInfo(name = "serialized")
    public String serialized;

    @ColumnInfo(name = "entry_id")
    public String entryId;

    @ColumnInfo(name = "answers")
    public List<Response> answers;

    @ColumnInfo(name = "selected_answer")
    public String selectedAnswer;

    @ColumnInfo(name = "type")
    public String type;
}

File: examples/android/CMeSdkJava/app_kt/src/main/kotlin/msdk/java/samplekt/db/entity/StructuredMessage.kt
package msdk.java.samplekt.db.entity

import androidx.room.ColumnInfo
import androidx.room.Entity

import androidx.room.PrimaryKey
import msdk.java.messages.StructuredMessage


@Entity
data class StructuredMessage(
        @PrimaryKey(autoGenerate = true)
        val id: Int = 0,

        @ColumnInfo(name = "pwdid")
        var pwDid: String,

        @ColumnInfo(name = "serialized")
        var serialized: String,

        @ColumnInfo(name = "entry_id")
        var entryId: String,

        @ColumnInfo(name = "answers")
        var answers: List<StructuredMessage.Response>,

        @ColumnInfo(name = "selected_answer")
        var selectedAnswer: String? = null,

        @ColumnInfo(name = "type")
        var type: String
)